pkg/cloudprovider/vsphere/loadbalancer/config/types_yaml.go

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2020 The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config
+
+/*
+	TODO:
+	When the INI based cloud-config is deprecated, this file should be renamed
+	from types_yaml.go to types.go and the structs within this file should be named:
+
+	LBConfigYAML -> LBConfig
+	LoadBalancerClassConfigYAML -> LoadBalancerClassConfig
+	LoadBalancerClassConfigYAML -> LoadBalancerClassConfig
+	NsxtConfigYAML -> NsxtConfig
+*/
+
+// LBConfigYAML  is used to read and store information from the cloud configuration file
+type LBConfigYAML struct {
+	LoadBalancer      LoadBalancerConfigYAML                  `yaml:"loadbalancer"`
```

Comment:
I wonder if we should use json tags here and use a json -> yaml serializer. That would give us json support for free while also supporting yaml :thinking:

Reply:
+1 on this

Reply:
Spoke with @andrewsykim about this in slack. I had a concern (not sure if it is a valid concern) with supporting multiple different configuration format types. We currently support INI (legacy) and then want to add YAML for some advanced configurations using YAML unlocks. The original idea was to have a uniform experience for k8s configuration and then the CPI. The concern is having multiple different formats INI, YAML, JSON and etc and supporting all the different flavors. It seems like it's more to go wrong especially when it comes to a user configuration/interface to the CPI.

If this isn't a valid concern or there is additional calls from the community, Andrew and I can always include the additional JSON support.